code/modules/mob/living/simple_animal/hostile/retaliate/fish.dm

```diff
@@ -27,4 +27,43 @@
 	speak_emote = list("gnashes")
 
 	faction = list("carp")
-	flying = 1
\ No newline at end of file
+	flying = 1
+
+
+/mob/living/simple_animal/hostile/retaliate/carp/koi
+	name = "space koi"
+	desc = "a gentle space faring koi."
+	icon = 'icons/obj/fish_items.dmi'
+	icon_state = "koi1"
+	icon_living = "koi1"
+	icon_dead = "koi1-dead"
+	icon_gib = "carp_gib"
+
+	harm_intent_damage = 1
+	melee_damage_lower = 2
+	melee_damage_upper = 2
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+	speak_emote = list("blurps")
+	butcher_results = null
+
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	faction = list("carp")
+	flying = 1
+	gold_core_spawnable = CHEM_MOB_SPAWN_FRIENDLY
```

Comment:
These should be hostile. They're absolutely not friendly.

Retaliate mobs will tear you to bits if you hit them even once; and not just the one that you hit.

Reply:
Gotcha

Reply:
While Fox is correct, the retaliate carp (and thus koi by inheritence) will attempt to retreat when attack, so these ones would much rather flee from you than fight, but will still fight back if trapped. Just posting this for clarification on their behavior.

Still make it a hostile classification for core/chem spawning though.